### PR TITLE
Run PHPCS in CI for FSE plugin

### PR DESCRIPTION
**Pull request by noahtallen**
#### Changes proposed in this Pull Request
* Run phpcs lint check on changed files in the FSE plugin
* Expand FSE plugin GH workflow to run on changes to yarn lock file and to apps/full-site-editing

#### Testing instructions
* Lint check should work and should fail on errors

I verified that this does the following:
1. Fails when a change to a PHP file is not formatted correctly
2. Does not try to lint non-php files
3. Runs on changes to yarn.lock

### CODE REVIEW

**Review by Addison-Stavlo — COMMENTED** [review]
This is great!  And the PR even tests itself from what it looks like? 😆 

Screen Shot 2020-06-18 at 4 30 47 PM

Screen Shot 2020-06-18 at 4 30 35 PM

File: .github/workflows/full-site-editing-plugin.yml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 on:
   pull_request:
-    # only trigger this workflow if FSE plugin files have been modified
+    # only trigger this workflow if FSE plugin files have been modified, or if packages have been updated.
     paths:
-    - 'apps/full-site-editing/full-site-editing-plugin/**'
+    - 'apps/full-site-editing/**'
```
Comment (vindl) [inline .github/workflows/full-site-editing-plugin.yml:5]: Not sure if we should build and sync diff creation for things outside of actual plugin folder. 🤔 It probably won't cause issues since we only merge version bump PRs anyway. Another option would be to isolate the changes here in a separate workflow file with these broader matchers.

Comment (noahtallen) [reply .github/workflows/full-site-editing-plugin.yml:5]: That might be possible. IMO, we should be running:
- `yarn build`
- `phpunit`
- `phpcs`

to any change in apps/full-site-editing (e.g. to pick up on config changes). And on changes to yarn-lock. 

IMO the sync diff is the exception 🤔 I wonder if we can restrict a job based on the path itself.

**Comment by ockham** [comment]
> This is great! And the PR even tests itself from what it looks like? 
> 
> Screen Shot 2020-06-18 at 4 30 47 PM
> 
> Screen Shot 2020-06-18 at 4 30 35 PM

Only way to test it :sunglasses:

File: .github/workflows/full-site-editing-plugin.yml (hunk 1 above)
Comment (ockham) [reply .github/workflows/full-site-editing-plugin.yml:5]: > Not sure if we should build and sync diff creation for things outside of actual plugin folder.  It probably won't cause issues since we only merge version bump PRs anyway.

Yeah, and there's currently not a lot of files in `apps/full-site-editing/` (outside of `apps/full-site-editing/full-site-editing-plugin`), so the overhead shouldn't be too bad.

> Another option would be to isolate the changes here in a separate workflow file with these broader matchers.

That might make sense in the long run.

Personally, I wouldn't block merging this PR on that -- I think it's important to get PHPCS into CI, and we can tweak and iterate _how_ we do it later.

File: yarn.lock [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. ars
```
Comment (ockham) [inline yarn.lock]: :peach:

**Review by ockham — APPROVED** [review]
Thanks! This is going to be very valuable for FSE.

There's been a few good suggestions on this PR, but I think it's important to get PHPCS into CI soon. We can then tweak it in later iterations.

File: .github/workflows/full-site-editing-plugin.yml (hunk 1 above)
Comment (vindl) [reply .github/workflows/full-site-editing-plugin.yml:5]: > IMO the sync diff is the exception 🤔 I wonder if we can restrict a job based on the path itself.

Afaik that hasn't been introduced in GH actions proper yet, although I've seen some 3rd party actions for it (I would rather avoid that dependency).

Comment (vindl) [reply .github/workflows/full-site-editing-plugin.yml:5]: I don't consider this a blocker either, just a possible future improvement.

**Review by vindl — APPROVED** [review]

File: yarn.lock (hunk 2 above)
Comment (noahtallen) [reply yarn.lock]: lol

File: .github/workflows/full-site-editing-plugin.yml [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
+    - 'yarn.lock'
```
Comment (vindl) [inline .github/workflows/full-site-editing-plugin.yml:6]: It turns out that this match is too broad and it will trigger on PRs that are not related to FSE plugin. For example see D45192-code.

Comment (ockham) [reply .github/workflows/full-site-editing-plugin.yml:6]: I'll file a PR to remove it :+1:

Comment (ockham) [reply .github/workflows/full-site-editing-plugin.yml:6]: #43480
